programs/local/LocalServer.cpp

```diff
@@ -148,6 +160,19 @@ static DatabasePtr createMemoryDatabaseIfNotExists(ContextPtr context, const Str
     return system_database;
 }
 
+static DatabasePtr createClickHouseLocalDatabaseOverlay(const String & name_, ContextPtr context_)
+{
+    auto databaseCombiner = std::make_shared<DatabasesOverlay>(name_, context_);
+    databaseCombiner->registerNextDatabase(std::make_shared<DatabaseFilesystem>(name_, "", context_));
+    databaseCombiner->registerNextDatabase(std::make_shared<DatabaseMemory>(name_, context_));
+#if USE_AWS_S3
+    databaseCombiner->registerNextDatabase(std::make_shared<DatabaseS3>(name_, "", "", context_));
```

Comment:
we need to do something with credentials, currently they are not supported.
also it does not seem right that we register s3 and hdfs databases even though a user might not need it, but we will query s3 and hdfs regardless.
may be register them only if some client option is added? like passing a config file under `--register_databases` option, and in this config file we will define s3, hdfs databases and credentials for them?
like 
```
<register_databases>
    <s3>
       <access_key_id></access_key_id>
       <secret_access_key></secret_access_key>
    </s3>
</register_databases>
```